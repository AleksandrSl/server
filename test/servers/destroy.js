#!/usr/bin/env node

'use strict'

const Server = require('../../server')

const app = new Server({
  subprotocol: '1.0.0',
  supports: '1.x',
  port: 2000
})
app.nodeId = 'server:FnXaqDxY5e'

app.auth(() => Promise.resolve(true))

app.unbind.push(() => new Promise(resolve => {
  setTimeout(() => {
    process.stderr.write(' Custom destroy task finished\n')
    resolve()
  }, 10)
}))

app.listen()

process.on('message', msg => {
  if (msg === 'close') {
    console.error('close')
    app.destroy()
  }
})
