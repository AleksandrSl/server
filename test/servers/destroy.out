
 INFO  1970-01-01 00:00:00 Logux server is listening
       Logux server: 0.0.0
       Node ID:      server
       Environment:  test
       Subprotocol:  1.0
       Supports:     1.x
       Listen:       ws://127.0.0.1:1337

 INFO  1970-01-01 00:00:00 Shutting down Logux server
 Custom destroy task finished
